lib/Driver/Driver.cpp

```diff
@@ -413,9 +413,21 @@ std::unique_ptr<Compilation> Driver::buildCompilation(
   bool ShowIncrementalBuildDecisions =
     ArgList->hasArg(options::OPT_driver_show_incremental);
 
-  bool Incremental = ArgList->hasArg(options::OPT_incremental) &&
-    !ArgList->hasArg(options::OPT_whole_module_optimization) &&
-    !ArgList->hasArg(options::OPT_embed_bitcode);
+  bool Incremental = ArgList->hasArg(options::OPT_incremental);
+  if (ArgList->hasArg(options::OPT_whole_module_optimization)) {
+    if (ShowIncrementalBuildDecisions) {
+      llvm::outs() << "Incremental compilation has been disabled in favor of "
+                   << "whole module optimization.";
+    }
+    Incremental = false;
+  }
+  if (ArgList->hasArg(options::OPT_embed_bitcode)) {
+    if (ShowIncrementalBuildDecisions) {
+      llvm::outs() << "Incremental compilation has been disabled in favor of "
+                   << "embedding LLVM IR bitcode.";
```

Comment:
This one seems funny for the same reason. (And we *could* support it; we just don't bother because no one uses this mode except to submit to the store, which is always a clean build.)

Also, if you use WMO *and* -embed-bitcode you'll get both messages. I guess you need both of them if you're trying to *get* to an incremental build, but in the normal case it seems redundant. What do you think?